<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="RobotControl" type="system">
        <hardware>
            <plugin>bicdrive_arduino/CarlikeBotSystemHardware</plugin>
            <param name="front_wheel_name">virtual_front_wheel_joint</param>
            <param name="rear_wheel_name">virtual_rear_wheel_joint</param>
            <param name="loop_rate">30</param>
            <param name="device">/dev/serial/by-path/pci-0000:00:14.0-usb-0:4:1.0-port0</param>
            <param name="baud_rate">9600</param>
            <param name="timeout_ms">1000</param>
        </hardware>

        <joint name="virtual_front_wheel_joint">
            <command_interface name="position"/>
            <state_interface name="position"/>
        </joint>
        
        <joint name="virtual_rear_wheel_joint">
            <command_interface name="velocity"/>
            <state_interface name="velocity"/>
        </joint>
    </ros2_control>
</robot>